tests/unit/lib/test_opensearch_exclusions.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import json
+import os
+import unittest
+from unittest.mock import PropertyMock, patch
+
+from charms.opensearch.v0.constants_charm import NodeLockRelationName, PeerRelationName
+from charms.opensearch.v0.helper_conf_setter import YamlConfigSetter
+from charms.opensearch.v0.models import DeploymentState, DeploymentType, State
+from charms.opensearch.v0.opensearch_internal_data import Scope
+from charms.opensearch.v0.opensearch_nodes_exclusions import VOTING_TO_DELETE
+from ops.testing import Harness
+
+from charm import OpenSearchOperatorCharm
+from tests.unit.helpers import mock_deployment_desc
+
+
+class TestOpenSearchExclusions(unittest.TestCase):
+    def setUp(self) -> None:
+        self.harness = Harness(OpenSearchOperatorCharm)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.set_leader(True)
+
+        self.harness.begin()
+
+        self.charm = self.harness.charm
+        self.peer_rel_id = self.harness.add_relation(PeerRelationName, self.charm.app.name)
+        self.lock_rel_id = self.harness.add_relation(NodeLockRelationName, self.charm.app.name)
```

Comment:
U may B able 2 get rid of this line.

Reply:
Only the last line. The peer rel is needed so I can set the deployment-description content later down the road.